### PR TITLE
Patch 33

### LINKED ISSUE #34: [`v2`] Initial Restructuring of the Package
For the new release of `v2a1` next month, as detailed in #33, an initial restructuring of the package must be done:
> ```
> DiffPriv
>  ├── scheme 
>  │    └── pre
>  │    └── utils
>  ├── mech
>  │     ├── pre
>  │     ├── validate
>  │     └── utils
>  ├── cli
>  └── utils
> ```
As detailed in #33:
>  - `scheme` - Contains the scheme class,
>    - `pre` - Contains pre-built encryption schemes
>   - `utils` - Encryption related utilities.
>  - `mech` - Contains the mechanism class,
>    - `pre` - Contains pre-built mechanisms
>    - `validate` - Validation of mechanism parameters.
>    - `utils` - Privacy related utilities.
>  - `utils` - Warnings, Errors, and other utilities for the package.
>  - `cli` - CLI interface
## Additional Notes
Work should be done on the `v2` branch.

### PR DESCRIPTION
**Pull request by quantum9Innovation**
This PR works to solve some of the issues mentioned in https://github.com/Quantalabs/DiffPriv/discussions/33. Please note that much of the code added in this release is highly experimental (and not quite tested either; hence the drop in code coverage). We hope to expand on this later before it is shipped into production.

---
See commit for more detailed information.

Closes #34

### CODE REVIEW

**Comment by quantum9Innovation** [comment]
Will get working on the code coverage issues as soon as possible. However, after the review process is done and if possible, this should be merged right into the v2 branch (after all, it is just an alpha release). @Quantalabs you may need to use admin privileges to accomplish this.

File: DiffPriv/mech/__init__.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+"""Mechanism base utils"""
+
+from .. import math
+
+from .main import *
+from . import validate
+from . import pre
+from . import utils
```
Comment (Quantalabs) [inline DiffPriv/mech/__init__.py:8]: Why not just move `main` here?

File: DiffPriv/mech/validate.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
+"""Validates mechanism params"""
+
+class DomainError(Exception):
```
Comment (Quantalabs) [inline DiffPriv/mech/validate.py]: These errors should be in `DiffPriv/utils.py`.

File: setup.py [hunk 3]
```diff
@@ -5,7 +5,7 @@
 
 setuptools.setup(
     name="DiffPriv",
-    version="v1.0.3",
+    version="v2.0.0",
```
Comment (Quantalabs) [inline setup.py]:
```suggestion
    version="v2.0.0-alpha1",
```

Comment (Quantalabs) [reply setup.py]: Or:
```suggestion
    version="v2.0.0a1",
```

**Review by Quantalabs — CHANGES_REQUESTED** [review]
Before merge:
- Regenerate documentation, and check all the docstrings are accurate- make adjustments as needed
- Fix suggestions listed below.

File: DiffPriv/mech/validate.py (hunk 2 above)
Comment (quantum9Innovation) [reply DiffPriv/mech/validate.py]: Doing this first (1)

**Comment by quantum9Innovation** [comment]
> Regenerate documentation, and check all the docstrings are accurate- make adjustments as needed

Doing this second (2)

**Comment by Quantalabs** [comment]
Why are you merging to main? Merge to https://github.com/Quantalabs/DiffPriv/tree/v2

**Comment by quantum9Innovation** [comment]
ETA for this PR is ~24 hours
Changes should be made around the same time tomorrow; after merging expect another week before code coverage and testing updates

**Comment by Quantalabs** [comment]
> Why are you merging to main? Merge to `v2`

Fixed

**Comment by Quantalabs** [comment]
@quantum9Innovation additionally, add tests for classes and functions. If any additional deepsource errors come up, fix those too.

**Comment by quantum9Innovation** [comment]
On it (3)

**Comment by quantum9Innovation** [comment]
## **Progress**

- [x] (1)
- [x] (2)
- [x] (3)

**Comment by quantum9Innovation** [comment]
Unfortunately delays have pushed back the merge date to today. PR should be ready to merge in a few hours.

File: DiffPriv/mech/validate.py (hunk 2 above)
Comment (quantum9Innovation) [reply DiffPriv/mech/validate.py]: Committing now ...

**Comment by quantum9Innovation** [comment]
> Before merge:
> 
> * Regenerate documentation, and check all the docstrings are accurate- make adjustments as needed
> * Fix suggestions listed below.

Done, although it seems accomplishing (3) is harder than it seems (and thus may take some more time). This should likely be postponed until the beta release. @Quantalabs any thoughts?

**Comment by quantum9Innovation** [comment]
> @quantum9Innovation additionally, add tests for classes and functions. If any additional deepsource errors come up, fix those too.

All 2 suggestions have been implemented, (3) is complete, and DeepSource is continuing to pass. Ready for merge after review from @Quantalabs

**Pull request merged**